carbon-design-system/carbon
# ci(release): remove deprecated action, call github release API directly

Closes https://github.com/carbon-design-system/carbon/issues/12500
v10 counterpart: https://github.com/carbon-design-system/carbon/pull/13018

#### Changelog

**Changed**

- Update the release workflow to directly call the GitHub API to create a new release

**Removed**

- Remove unmaintained `create-release` action

#### Testing / Reviewing

- Make sure the syntax looks correct for using the [`github-script` action](https://github.com/actions/github-script)
- Make sure I'm not leaving out any required fields for the [`createRelease` API endpoint](https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#create-a-release)
